### PR TITLE
Update the API-Documentation ADR to reflect the latest swagger changes

### LINKED ISSUE #188: Update the API-Documentation ADR to reflect the latest swagger updates

### PR DESCRIPTION
**Pull request by Aymalla**
closes #188

### CODE REVIEW

File: data-reconciliation-app/docs/adr/09-api-documentation.md [hunk 1]
```diff
@@ -20,22 +20,26 @@ This document can be imported into [`Swagger`](https://editor.swagger.io/) to ge
...
+Create two new endpoints as part of the application API
+ - /app/swagger.json: return an OpenAPI documentation in json formate.
+ - /app/swagger: return the html for swagger-ui which will a UI based on the OpenAPI documentation
 
-Create a new separate backend application that returns an OpenAPI documentation UI (swagger-UI), 
-
-- The application can act as a wrapper on top of the CCF OpenAPI endpoint to retrieve the document, convert it to UI, and return the result, and it can be written in .NET, Node.js, or any other language.
-
-- The application will generate a UI using a separate copy of the openAPI document defined within the application boundary (this document must be maintained alongside the endpoints).
+Note: Currently, CCF `/app/api` OpenAPI documentation endpoint implementation lacks the methodology for deploying the different options supported by OpenAPI specifications as a part of the application bundle `app.json`, like (info, schemes, security definitions, and host), when the issues are solved, the `/app/swagger.json` will be replaced by the built-in ccf endpoint `/app/api`
```
Comment (ross-p-smith) [inline data-reconciliation-app/docs/adr/09-api-documentation.md:29]: Can we link this to an issue in ccp repo?

**Review by ross-p-smith — APPROVED** [review]
Thanks for updating the ADR!

File: data-reconciliation-app/docs/adr/09-api-documentation.md [hunk 2]
```diff
@@ -20,22 +20,26 @@ This document can be imported into [`Swagger`](https://editor.swagger.io/) to ge
...
 
 ## Decision
 
-We will go with the first option because:
-- It has the least development and maintenance effort.
-- It will evolve and be updated automatically because it is leveraging a CCF built-in feature.
+We will go with the second option because:
+- Gives a better developer experience to document, explore and test the application endpoints.
+- Will evolve and be updated automatically because it is part of the application.
+- leverage the built-in features for swagger-ui like authentication using JWT or mutual TLS Certs
```
Comment (macromania) [inline data-reconciliation-app/docs/adr/09-api-documentation.md:36]: Nit: 
```suggestion
- Leverage the built-in features for swagger-ui like authentication using JWT or mTLS 
```

File: data-reconciliation-app/docs/adr/09-api-documentation.md (hunk 1 above)
Comment (Aymalla) [reply data-reconciliation-app/docs/adr/09-api-documentation.md:29]: I've added a new [issue](https://github.com/orgs/microsoft/projects/542/views/1?pane=issue&itemId=19425859) in our Repo and will discuss it with Amaury tomorrow, before convert it to issue on the Main CCF Repo

**Pull request merged**